### PR TITLE
Implementing Sort in  digest and tests for it

### LINKED ISSUE #7390: Sorting does not work in digest
### Please describe the problem (or idea)

> What happened just before the problem occurred? Or what problem could this idea solve?
Sorting doe not work in digest
**Current Behaviour**
sorting

> What did you expect to see that you didn't?

### Please show us where to look

https://publiclab.org/ ...

### What's your PublicLab.org username?

> This can help us diagnose the issue: 

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by sssash18**
Fixes #7390  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below
**New Behaviour**
Untitled_ Feb 13, 2020 5_32 PM

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by sssash18** [comment]
Restarting travis

**Comment by sssash18** [comment]
> This looks good! Maybe since it's a new feature we could add a controller test for this functionality? Thanks

Yes @Uzay-G  I would add the test. Should I do it in a separate PR or I should add it here itself?

**Comment by Uzay-G** [comment]
> > This looks good! Maybe since it's a new feature we could add a controller test for this functionality? Thanks
> 
> Yes @Uzay-G I would add the test. Should I do it in a separate PR or I should add it here itself?

As you prefer :+1:

**Comment by sssash18** [comment]
@Uzay-G Should I modify the nodes.yml file as for example all the nodes in digest have same(=0) cached likes .

**Comment by Uzay-G** [comment]
> @Uzay-G Should I modify the nodes.yml file as for example all the nodes in digest have same(=0) cached likes .

Yeah that's no problem :+1:

**Comment by Uzay-G** [comment]
So @sssash18 are you going to open a new pr for the tests or do we stay on this one?

**Comment by sssash18** [comment]
@Uzay-G I will make a new PR for the tests.Thanks.

**Comment by cesswairimu** [comment]
@sssash18 could ypu just add the tests on this branch so that we can merge them together? Thanks

**Comment by sssash18** [comment]
> @sssash18 could ypu just add the tests on this branch so that we can merge them together? Thanks

Ok @cesswairimu .

**Comment by sssash18** [comment]
@Uzay-G @cesswairimu some of the other tests are failing after making changes in nodes.yml. Please review

**Comment by sssash18** [comment]
@Uzay-G @cesswairimu some of the other tests are failing after making changes in nodes.yml. Please review

**Comment by Uzay-G** [comment]
For some reason this test is failing:

https://github.com/publiclab/plots2/blob/db94d3cc7ad06fdd6f989927ee52ccd8b8df6ad8/test/functional/like_controller_test.rb#L22-L38

**Comment by Uzay-G** [comment]
Yeah this same error seems to be affecting the unit tests.

**Comment by sssash18** [comment]
> For some reason this test is failing:
> 
> https://github.com/publiclab/plots2/blob/db94d3cc7ad06fdd6f989927ee52ccd8b8df6ad8/test/functional/like_controller_test.rb#L22-L38

@Uzay-G If i change '1' to '31' in line 32 then it would be fine. What to do you think?

**Comment by Uzay-G** [comment]
That number represents the number of likes of the note right? Then, to have more maintainable code, change '1' to `note.cached_likes`

**Comment by sssash18** [comment]
@Uzay-G I found that the value of note.cached_likes always return 1 . Please check it .

**Comment by cesswairimu** [comment]
Okay I am looking into the cached_likes issue..Also to make the tests better we shouldn't include specific assertions like `assert_equal(wikis.first.title, "Canon A1200 IR conversion at PLOTS Barnraising at LUMCON")`  because if we ever add another wiki that  has a title starting with `B` or `A` this test will fail...Does this make sense?  
We can try something like  
```
test 'sorting of digest by title' do
     UserSession.create((users(:bob)))
     get :digest, params: { sort: "title"}
    assert assigns(:wikis).each_cons(2).all?{|i,j| j.title >= i.title }
end
```
What do you think?

**Comment by cesswairimu** [comment]
You can check out how we do it in tags test  https://github.com/publiclab/plots2/blob/master/test/functional/tag_controller_test.rb#L202-L215 Let me know if you need any help. Thanks

**Comment by sssash18** [comment]
> Okay I am looking into the cached_likes issue..Also to make the tests better we shouldn't include specific assertions like `assert_equal(wikis.first.title, "Canon A1200 IR conversion at PLOTS Barnraising at LUMCON")` because if we ever add another wiki that has a title starting with `B` or `A` this test will fail...Does this make sense?
> We can try something like
> 
> ```
> test 'sorting of digest by title' do
>      UserSession.create((users(:bob)))
>      get :digest, params: { sort: "title"}
>     assert assigns(:wikis).each_cons(2).all?{|i,j| j.title >= i.title }
> end
> ```
> 
> What do you think?

Yes @cesswairimu , It's a good idea. I will implement it soon .Thanks for your advice

**Comment by sssash18** [comment]
@ cesswairimu I updated the tests , please have a look on failing travis tests.

**Comment by cesswairimu** [comment]
@sssash18 could please try rebasing? see if the travis will pass?

**Comment by sssash18** [comment]
@cesswairimu I rebased , but their are some other tests which still fails. Please have a look!

File: test/functional/like_controller_test.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def teardown; end
     assert_response :success
 
     note = Node.find note.id
-    assert_equal @response.body, '1'
```
Comment (Uzay-G) [inline test/functional/like_controller_test.rb]: why did u change this to an int? One of the tests is faling because it expects "1" and not 1.

**Comment by Uzay-G** [comment]
@sssash18 the tests will be resolved once you fix the comment i made on the pr.

**Comment by sssash18** [comment]
@Uzay-G I rectified that. Additionally I noticed that the like_controller_test failed due to line 34 , while testing on my local machine. But line 34 is already substituted with `note.cached_likes` .Please guide on that. Thanks!

**Pull request closed without merging**